src/error.rs

```diff
@@ -135,6 +153,35 @@ impl From<serde_json::Error> for Error {
     }
 }
 
+impl From<bs58::decode::Error> for Error {
+    fn from(err: bs58::decode::Error) -> Self {
+        match err {
+            bs58::decode::Error::BufferTooSmall => {
+                Error::msg(ErrorKind::InvalidState, err.to_string())
+            }
+            bs58::decode::Error::InvalidCharacter {
+                character: _,
+                index: _,
+            } => Error::msg(ErrorKind::InvalidState, err.to_string()),
+            bs58::decode::Error::NonAsciiCharacter { index: _ } => {
+                Error::msg(ErrorKind::InvalidState, err.to_string())
+            }
+            _ => Error::msg(ErrorKind::Malformed, err.to_string()),
+        }
+    }
+}
+
+impl From<bs58::encode::Error> for Error {
+    fn from(err: bs58::encode::Error) -> Self {
+        match err {
+            bs58::encode::Error::BufferTooSmall => {
+                Error::msg(ErrorKind::InvalidState, err.to_string())
+            }
+            _ => Error::msg(ErrorKind::Malformed, err.to_string()),
```

Comment:
I believe we should throw Malformed only when decoding / de-serializing, that is when we have some output that we don't control. 
When encoding / serializing, we don't expect any "malformed" input, as we created that input in our code. So, if something unexpected happens during encoding / serialization, that is probably always `InvalidState`, not `Malformed`.

Reply:
fixed